channel/state/state.go

```diff
@@ -165,3 +166,24 @@ func (s State) RecoverSigner(sig Signature) (types.Address, error) {
 	}
 	return RecoverEthereumMessageSigner(stateHash[:], sig)
 }
+
+// Clone returns a clone of the state
+func (s State) Clone() State {
+	clone := State{}
+
+	// shallow clone of fixed part
```

Comment:
Is there an easy way for me to know that the shallow clone here is a safe clone?

Reply:
These are shallow clones of values that _should never change_.